sound/soc/sof/core.c

```diff
@@ -292,6 +292,31 @@ static int sof_machine_check(struct snd_sof_dev *sdev)
 	return 0;
 #endif
 }
+EXPORT_SYMBOL(sof_machine_check);
+
+
+int sof_machine_register(struct snd_sof_dev *sdev, void *pdata)
+{
+	struct snd_sof_pdata *plat_data = (struct snd_sof_pdata*)pdata;
+	const char *drv_name;
+	const void *mach;
+	int size;
+
+	drv_name = plat_data->machine->drv_name;
+	mach = (const void *)plat_data->machine;
+	size = sizeof(*plat_data->machine);
+
+	/* register machine driver, pass machine info as pdata */
+	plat_data->pdev_mach =
+		platform_device_register_data(sdev->dev, drv_name,
+					      PLATFORM_DEVID_NONE, mach, size);
+	if (IS_ERR(plat_data->pdev_mach))
+		return PTR_ERR(plat_data->pdev_mach);
+
+	dev_dbg(sdev->dev, "created machine %s\n",
+		dev_name(&plat_data->pdev_mach->dev));
+}
+EXPORT_SYMBOL(sof_machine_register);
```

Comment:
if we want to be consistent, we need an sof_machine_unregister(), to be called from line 524

Reply:
@plbossart fixed.